micro-rdk/src/common/config_monitor.rs

```diff
@@ -108,7 +108,7 @@ where
                     }
                 }
 
-                if *config != *self.curr_config {
+                if config.revision != self.curr_config.revision {
```

Comment:
should we store the revision in NVS separately so that we don't have to load the whole config for this comparison?

Reply:
yeah that would be better, and is how the OTA check works.

Reply:
I'm not sure I like that idea, since it means we have the same information stored in two places, which means there is the opportunity for the two to desynchronize. However, it may be possible to reduce the memory requirements for this class by storing only the config revision rather than a copy of the whole config.